src/gsi/read_gnssrspd.f90

```diff
@@ -0,0 +1,515 @@
+           else
+              if (abs(tdiff)>twind) cycle loop_readsb2
+           endif
+           nread = nread+1
+
+           usage = zero                ! will be considered for assimilation
+                                       ! subject to further QC in setupt subroutine
+           iuse  = icuse(nc)           ! assimilation flag 
+           if (iuse <=0) usage = r100  ! will be monitored but not assimilated
+
+!          Read observation location (lat/lon degree) 
+           call ufbint(lunin,obsloc,2,1,nlv,locstr)
+
+           if (obsloc(1,1) == missing .or. abs(obsloc(1,1)) < -180.0_r_kind .or. &     
+               obsloc(1,1) == missing .or. abs(obsloc(1,1)) >  180.0_r_kind .or. &
+               obsloc(2,1) == missing .or. abs(obsloc(2,1)) <  -90.0_r_kind .or. &
+               obsloc(2,1) == missing .or. abs(obsloc(2,1)) >   90.0_r_kind ) then 
+               write(6,*) 'READ_GNSSRSPD: bad lon/lat values: ', obsloc(1,1),obsloc(2,1)              
+               cycle loop_readsb2     
+           endif
+! GNSSRSPD BUFR longitudes are in the +/- 180 range, need to convert to 0 to 360
+! deg
+           if (obsloc(1,1) < 0.0_r_kind) obsloc(1,1) = obsloc(1,1) + 360.0_r_kind
+           !if (obsloc(1,1) >= 0.0_r_kind .or. obsloc(1,1) <= 180.0_r_kind) obsloc(1,1) = obsloc(1,1) + 180.0_r_kind
+ 
+           dlon_earth_deg = obsloc(1,1)
+           dlat_earth_deg = obsloc(2,1)
+           dlon_earth = obsloc(1,1)*deg2rad ! degree to radian
+           dlat_earth = obsloc(2,1)*deg2rad ! degree to radian
+
+!          Convert obs lat/lon to rotated coordinate and check 
+!          if the obs is outside of the regional domain
+           if (regional) then
+              call tll2xy(dlon_earth,dlat_earth,dlon,dlat,outside)
+              if (diagnostic_reg) then
+                 call txy2ll(dlon,dlat,rlon00,rlat00)
+                 ntest      = ntest+1
+                 cdist      = sin(dlat_earth)*sin(rlat00)+cos(dlat_earth)*cos(rlat00)* &
+                             (sin(dlon_earth)*sin(rlon00)+cos(dlon_earth)*cos(rlon00))
+                 cdist      = max(-one,min(cdist,one))
+                 disterr    = acos(cdist)*rad2deg
+                 disterrmax = max(disterrmax,disterr)
+              end if
+              if(outside) cycle loop_readsb2
+           else
+              dlon = dlon_earth
+              dlat = dlat_earth
+              call grdcrd1(dlat,rlats,nlat,1)
+              call grdcrd1(dlon,rlons,nlon,1)
+           endif
+
+
+!          Read surface wind speed [m/s] from GNSSRSPD 
+           if (lspdob) then
+              !usage=r100 
+              !usage = zero  ! will be considered for assimilation
+
+              ! Get Wind Speed observations from bufr file
+              call ufbint(lunin,gnssrw,1,1,nlv,wndstr)
+              spdob = gnssrw(1,1) ! surface wind speed 
+
+              ! Don't permit observations with ws <= 1 m/s
+              if (spdob <= 1.0) cycle loop_readsb2
```

Comment:
Append `_r_kind` to `1.0` or use constant `one`

Reply:
Done!